File: scripts/_capture_pratinjau.py
import time, os, json, hmac, base64, hashlib
from playwright.sync_api import sync_playwright

OWNER_ID = "cmt2bwnfe000g4o72n2oa2qjx"
SECRET = os.environ.get("SESSION_SECRET") or os.environ.get("CRON_SECRET")
BASE = "http://localhost:3100"
OUT = "dogfood-output/pratinjau"
MAX_AGE = 60 * 60 * 24 * 30

def make_token(uid):
    exp = int(time.time() * 1000) + MAX_AGE * 1000
    body = base64.urlsafe_b64encode(f"{uid}.{exp}".encode()).rstrip(b"=").decode()
    sig = hmac.new(SECRET.encode(), body.encode(), hashlib.sha256).hexdigest()
    return f"{body}.{sig}"

SCREENS = [
    ("/app", "01-dashboard", "Dashboard Ringkas", "Pantau pekerjaan, pengingat & omzet dari satu layar."),
    ("/app/pelanggan", "02-pelanggan", "Daftar Pelanggan", "Semua pelanggan tercatat rapi, siap dihubungi."),
    ("/app/pekerjaan", "03-pekerjaan", "Kelola Pekerjaan", "Jadwalkan & pantau pekerjaan teknisi real-time."),
    ("/app/layanan", "04-layanan", "Katalog Layanan & Harga", "Harga konsisten, termasuk harga khusus pelanggan."),
    ("/app/faktur", "05-faktur", "Invoice & Proforma", "Faktur profesional otomatis, kirim lewat WhatsApp."),
    ("/app/laporan", "06-laporan", "Laporan Keuangan", "Piutang, penerimaan, dan insentif teknisi otomatis."),
    ("/app/teknisi", "07-teknisi", "Kelola Tim Teknisi", "Atur teknisi & lihat performa masing-masing."),
    ("/app/unit", "08-unit", "Unit AC Pelanggan", "Riwayat tiap unit AC lengkap dengan jadwal servis."),
]

os.makedirs(OUT, exist_ok=True)
tok = make_token(OWNER_ID)
manifest = []
with sync_playwright() as p:
    b = p.chromium.connect_over_cdp("http://127.0.0.1:9222")
    ctx = b.new_context(viewport={"width":402,"height":874}, device_scale_factor=2, is_mobile=True)
    ctx.add_cookies([{"name":"aircon_tech","value":tok,"url":BASE}])
    pg = ctx.new_page()
    errs=[]; pg.on("pageerror", lambda e: errs.append(str(e)))
    for route, file, title, caption in SCREENS:
        try:
            r = pg.goto(f"{BASE}{route}", wait_until="networkidle", timeout=30000)
            pg.wait_for_timeout(1200)
            st = r.status if r else 0
            pg.screenshot(path=f"{OUT}/{file}.png", full_page=True)
            blen = len(pg.inner_text("body"))
            manifest.append({"route":route,"file":f"{file}.png","title":title,"caption":caption,"status":st,"bodyLen":blen})
            print(f"  {file}: HTTP {st}, teks {blen} char")
        except Exception as e:
            print(f"  {file}: GAGAL {str(e)[:80]}")
            manifest.append({"route":route,"file":f"{file}.png","title":title,"caption":caption,"error":str(e)[:120]})
    open(f"{OUT}/manifest.json","w").write(json.dumps(manifest, indent=2))
    print("exceptions:", len(errs))
    ctx.close()

File: scripts/_capture_pratinjau.mjs
/**
 * Capture layar aplikasi (owner demo) untuk isi Pratinjau — headless, viewport HP.
 * Mint cookie sesi (aircon_tech) utk owner demo (jalur getServerContext menerima user ACTIVE by id).
 * Hasil PNG di dogfood-output/pratinjau/. Anda tinggal unggah lewat /admin/landing → Pratinjau.
 */
import crypto from "crypto";
import { chromium } from "playwright";
import fs from "fs";

const OWNER_ID = "cmt2bwnfe000g4o72n2oa2qjx";
const SECRET = process.env.SESSION_SECRET || process.env.CRON_SECRET;
const BASE = "http://localhost:3100";
const OUT = "dogfood-output/pratinjau";
const MAX_AGE = 60 * 60 * 24 * 30;

function makeToken(userId) {
  const exp = Date.now() + MAX_AGE * 1000;
  const body = Buffer.from(`${userId}.${exp}`).toString("base64url");
  const sig = crypto.createHmac("sha256", SECRET).update(body).digest("hex");
  return `${body}.${sig}`;
}

// Layar + judul/keterangan (untuk nama file & referensi caption)
const SCREENS = [
  ["/app", "01-dashboard", "Dashboard Ringkas", "Pantau pekerjaan, pengingat & omzet dari satu layar."],
  ["/app/pelanggan", "02-pelanggan", "Daftar Pelanggan", "Semua pelanggan tercatat rapi, siap dihubungi."],
  ["/app/pekerjaan", "03-pekerjaan", "Kelola Pekerjaan", "Jadwalkan & pantau pekerjaan teknisi real-time."],
  ["/app/layanan", "04-layanan", "Katalog Layanan & Harga", "Harga konsisten, termasuk harga khusus pelanggan."],
  ["/app/faktur", "05-faktur", "Invoice & Proforma", "Faktur profesional otomatis, kirim lewat WhatsApp."],
  ["/app/laporan", "06-laporan", "Laporan Keuangan", "Piutang, penerimaan, dan insentif teknisi otomatis."],
  ["/app/teknisi", "07-teknisi", "Kelola Tim Teknisi", "Atur teknisi & lihat performa masing-masing."],
  ["/app/unit", "08-unit", "Unit AC Pelanggan", "Riwayat tiap unit AC lengkap dengan jadwal servis."],
];

(async () => {
  fs.mkdirSync(OUT, { recursive: true });
  const browser = await chromium.launch();
  const ctx = await browser.newContext({
    viewport: { width: 402, height: 874 }, // HP modern
    deviceScaleFactor: 2,
    isMobile: true,
  });
  await ctx.addCookies([{ name: "aircon_tech", value: makeToken(OWNER_ID), url: BASE }]);
  const pg = await ctx.newPage();
  const errs = [];
  pg.on("pageerror", (e) => errs.push(String(e)));

  const manifest = [];
  for (const [route, file, title, caption] of SCREENS) {
    try {
      const resp = await pg.goto(`${BASE}${route}`, { waitUntil: "networkidle", timeout: 30000 });
      await pg.waitForTimeout(1200);
      const status = resp?.status();
      // full page agar seluruh isi layar tertangkap
      await pg.screenshot({ path: `${OUT}/${file}.png`, fullPage: true });
      const bodyLen = (await pg.innerText("body")).length;
      manifest.push({ route, file: `${file}.png`, title, caption, status, bodyLen });
      console.log(`  ${file}: HTTP ${status}, teks ${bodyLen} char`);
    } catch (e) {
      console.log(`  ${file}: GAGAL ${e.message}`);
      manifest.push({ route, file: `${file}.png`, title, caption, error: e.message });
    }
  }
  fs.writeFileSync(`${OUT}/manifest.json`, JSON.stringify(manifest, null, 2));
  console.log("exceptions:", errs.length);
  console.log("MANIFEST:", `${OUT}/manifest.json`);
  await browser.close();
})();
